lib/Models/BaseMaps/BaseMapsModel.ts

```diff
@@ -90,28 +90,22 @@ export class BaseMapsModel extends CreateModel(BaseMapsTraits) {
     if (items !== undefined) {
       const { items: itemsTrait } = this.traits;
       const newItemsIds = itemsTrait.fromJson(this, stratumId, items);
-      newItemsIds
-        .catchError(error => {
-          errors.push(error);
-        })
-        ?.forEach((member: BaseMapModel) => {
-          const existingItem = this.items.find(
-            baseMap => baseMap.item === member.item
-          );
-          if (existingItem) {
-            // object array trait doesn't automatically update model item
-            existingItem.setTrait(stratumId, "image", member.image);
-          } else {
-            this.add(stratumId, member);
-          }
-        });
+      newItemsIds.pushErrorTo(errors)?.forEach((member: BaseMapModel) => {
+        const existingItem = this.items.find(
+          baseMap => baseMap.item === member.item
+        );
+        if (existingItem) {
+          // object array trait doesn't automatically update model item
+          existingItem.setTrait(stratumId, "image", member.image);
+        } else {
+          this.add(stratumId, member);
+        }
+      });
     }
 
-    updateModelFromJson(this, stratumId, rest).catchError(error => {
-      errors.push(error);
-    });
+    updateModelFromJson(this, stratumId, rest).pushErrorTo(errors);
 
-    return Result.return(
+    return new Result(
```

Comment:
could use a Result.error here to make a bit more explicit

Reply:
`Result.error()` always returns a `Result` with a `TerriaError`:

https://github.com/TerriaJS/terriajs/blob/842ffdaae2cff019632e260d173e8fb99f21899d/lib/Core/Result.ts#L84-L93

Whereas the constructor will only return a `Result` with a `TerriaError` if the `error` argument is defined:

https://github.com/TerriaJS/terriajs/blob/842ffdaae2cff019632e260d173e8fb99f21899d/lib/Core/Result.ts#L120-L128

In this example, it is using `TerriaError.combine()` - which may return no error (eg if `errors.length === 0`):

https://github.com/TerriaJS/terriajs/blob/5f299fb6747f84faf7d00b60bf42711f5085ad06/lib/Models/BaseMaps/BaseMapsModel.ts#L109-L115

Reply:
ah i see, the `Result(undefined, ...)` threw me off